File: CS/DashboardMerger/MergerBase.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace DashboardMerger {
    public static class NamesGenerator {
        public static string GenerateName(string name, int index, IEnumerable<string> occupiedNames) {
            string result = String.Format("{0}_{1}", name, index);
            if(occupiedNames.Contains(result))
                return GenerateName(name, ++index, occupiedNames);
            return result;
        }
    }
}

File: VB/DashboardMerger/MergerBase.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq

Namespace DashboardMergeExample
    Public NotInheritable Class NamesGenerator

        Private Sub New()
        End Sub

        Public Shared Function GenerateName(ByVal name As String, ByVal index As Integer, ByVal occupiedNames As IEnumerable(Of String)) As String
            Dim result As String = String.Format("{0}_{1}", name, index)
            If occupiedNames.Contains(result) Then
                index += 1
                Return GenerateName(name, index, occupiedNames)
            End If
            Return result
        End Function
    End Class
End Namespace
